### ISSUE TITLE
Code completion suggestions list won't converge 

### ISSUE THREAD

**Issue by ralfbiedert** [intellij-rust/intellij-rust#8074]
## Environment

* **IntelliJ Rust plugin version:** 0.4.158.4203-212
* **Rust toolchain version:** 1.58.0-nightly (00d5e42e7 2021-10-24) x86_64-pc-windows-msvc
* **IDE name and version:** CLion 2021.2.3 (CL-212.5457.51)
* **Operating system:** Windows 10 10.0
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description

Previously when invoking code completion I get some suggestions right away, with some others flying in about 500ms later. 

Since the latest update it appears that the suggestion list often won't _converge_ anymore, that is, suggestions appear and disappear for as long as the list is up:

https://user-images.githubusercontent.com/301288/140330671-6d808956-8eef-41fd-a6a1-8944e0308b6e.mp4

This happens even if all macros are expanded (#8020), with no "Analyzing" present and more conservative "Experimental Features" settings:

![image](https://user-images.githubusercontent.com/301288/140331900-79889f88-946d-4cd4-9553-d10870121d9e.png)

I've also tried the nightly Rust plugin 2 days ago and had similar behavior, but ran into other issues which is why I had to roll back.

## Steps to reproduce

Unfortunately I still don't have a good public repo to reproduce.

**Comment by ralfbiedert** [comment]
Two observations:

- With the latest `0.4.159.4232-212` release this got better, but I'm still running into it sometimes
- There seems to be some correlation with invoking this the first time on broken projects. I just used CLion for an hour without issues but had to restart it mid-coding on a function looking vaguely like 

```rust
    #[ffi_service_method(on_panic = "return_default")]
    pub fn collider_labels(&self) -> FFIOption<> {
        self.world_view.
    }
```

Navigating to `self.world_view.` and invoking auto complete could reproduce it, until I fixed the signature and replaced the body with `todo!()`. That seems to have made it recover and I could invoke it again on the original line.

**Comment by ralfbiedert** [comment]
This seems to be fixed with 2021.3 / 161.
